[Docs] Update determinism section (#2131)
ros-controls/ros2_control: controller_manager/doc/userdoc.rst

The limits will be applied after you log out and in again.

> You can run ros2_control with real-time requirements also from a docker container. Pass the following capability options to allow the container to set the thread priority and lock memory, e.g., for the ``ros2_control_demos`` container:
> 
> .. code-block:: console
> 
>     $ docker run -it \
>         --cap-add=sys_nice \
>         --ulimit rtprio=99 \
>         --ulimit memlock=-1 \
>         --rm --net host ros2_control_demos

Author: ```suggestion
You can run ros2_control with real-time requirements also from a docker container. Pass the following capability options to allow the container to set the thread priority and lock memory, e.g.,

.. code-block:: console

    $ docker run -it \
        --cap-add=sys_nice \
        --ulimit rtprio=99 \
        --ulimit memlock=-1 \
        --rm --net host <IMAGE>
```

>         --rm --name ros2_control_demos --net host ros2_control_demos

Author: this is the image name from the ros2_control_demo installation section?

>         --rm --name ros2_control_demos --net host ros2_control_demos

Reviewer: Yes, having container name and docker image name same might confuse someone new to dockers

>         --rm --name ros2_control_demos --net host ros2_control_demos

Author: the name is optional right? we can remove that also here https://github.com/ros-controls/ros2_control_demos/blob/master/doc/index.rst#using-docker

>         --rm --name ros2_control_demos --net host ros2_control_demos

Author: ```suggestion
        --rm --net host ros2_control_demos
```

For more information, see the Docker engine documentation about `resource_constraints <https://docs.docker.com/engine/containers/resource_constraints/#configure-the-real-time-scheduler>`__ and `linux capabilities <https://docs.docker.com/engine/containers/run/#runtime-privilege-and-linux-capabilities>`__.

The normal linux kernel is optimized for computational throughput and therefore is not well suited for hardware control.
Alternatives to the standard kernel include

> - `Real-time Ubuntu <https://ubuntu.com/real-time>`_ on Ubuntu 24.04 (also for RaspberryPi)

Author: the product of canonical is called _Real-time Ubuntu_

> - `Real-time Ubuntu <https://ubuntu.com/real-time>`_ on Ubuntu 24.04 (also for RaspberryPi)

Reviewer: ```suggestion
- `Real-time Ubuntu <https://ubuntu.com/real-time>`_ on Ubuntu (also for RaspberryPi)
```

Then let's remove the versioning here

robot_description [std_msgs::msg::String]

>   String with the URDF xml, e.g., from ``robot_state_publisher``.

Author: ```suggestion
  String with the URDF xml, e.g., from ``robot_state_publisher``.
  Reloading of the URDF is not supported yet.
  All joints defined in the ``<ros2_control>``-tag have to be present in the URDF.
```

  Reloading of the URDF is not supported yet.

>   All joints defined in the ``<ros2_control>``-tag have to be present in the URDF.

Author: This change was not intended here, idk
